crates/walrus-service/src/node/pending_metadata_cache.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Walrus Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! In-memory cache for metadata awaiting blob registration.
+
+use std::{
+    collections::HashMap,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+
+use tokio::sync::RwLock;
+use walrus_core::{BlobId, metadata::VerifiedBlobMetadataWithId};
+
+use super::metrics::NodeMetricSet;
+
+#[derive(Debug, Clone)]
+struct CachedMetadata {
+    inserted_at: Instant,
+    metadata: Arc<VerifiedBlobMetadataWithId>,
+}
+
+#[derive(Debug)]
+struct PendingMetadataCacheInner {
+    max_entries: usize,
+    ttl: Duration,
+    entries: HashMap<BlobId, CachedMetadata>,
+}
+
+impl PendingMetadataCacheInner {
+    fn new(max_entries: usize, ttl: Duration) -> Self {
+        Self {
+            max_entries,
+            ttl,
+            entries: HashMap::new(),
+        }
+    }
+
+    fn evict_expired(&mut self, now: Instant) {
+        if self.ttl.is_zero() {
```

Comment:
Question: why not evicting if ttl is zero? Isn't ttl == 0 indicate that the cache is disabled?